src/token.rs

```diff
@@ -0,0 +1,58 @@
+#[cfg(feature = "std")]
+use num_format::{CustomFormat, ToFormattedString};
+
+/// A given amount of token. Can be used for nicely formatted output and token-aware comparison of
+/// different amounts.
+///
+/// ```
+/// # use ss58_registry::Token;
+/// # #[cfg(feature = "std")]
+/// # fn x() {
+/// let my_token = Token { name: "I❤U", decimals: 8, amount: 100_000_000_000 };
+/// assert_eq!(format!("{}", my_token), "1_000,000 I❤U");
+/// assert_eq!(format!("{:?}", my_token), "1000,000 I❤U (100_000_000_000)");
+/// # }
+/// # #[cfg(not(feature = "std"))]
+/// # fn x() {}
+/// # x();
+/// ```
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Token {
+	/// The short name (ticker) of the token
+	pub name: &'static str,
+	/// The number of decimals the token has (smallest granularity of the token)
+	pub decimals: u8,
+	/// The amount in the smallest granularity of the token.
+	pub amount: u128,
+}
+
+#[cfg(feature = "std")]
+impl std::fmt::Display for Token {
```

Comment:
we could also have some test for this to make sure the details (number of decimal places etc.) not accidentally broken.

Reply:
The only "definition" of the `Display` format I found in the previous code I turned into a doctest. I am quite surprised the German decimal comma was hardwired into the format. Also, the thousands-separator `,` in the [sub-token code](https://github.com/paritytech/substrate-debug-kit/blob/9d31da7f01dfc8a6f3cd38c11a4e2cdff6279371/sub-tokens/src/lib.rs#L17-L18) was colliding with the decimal comma, so I made a brave move and replaced the thousands-separator with `_`.

I can move the doctests into regular tests if you prefer them, although they [are run by the CI](https://gitlab.parity.io/parity/ss58-registry/-/jobs/1421296#L189).